src/unix/kqueue.c

```diff
@@ -621,10 +621,12 @@ int uv_fs_event_start(uv_fs_event_t* handle,
 #endif /* #if defined(__APPLE__) && MAC_OS_X_VERSION_MAX_ALLOWED >= 1070 */
 
   uv__handle_start(handle);
-  uv__io_init(&handle->event_watcher, uv__fs_event, fd);
-  uv__io_start(handle->loop, &handle->event_watcher, POLLIN);
 
-  return 0;
+  return uv__io_init_start(handle->loop,
+                           &handle->event_watcher,
+                           uv__fs_event,
+                           fd,
+                           POLLIN);
```

Comment:
If this fails, then the effects of the `uv_handle_start()` call right above it should be undone.

The other way around is even better because more efficient: only call `uv_handle_start()` when `uv__io_init_start()` succeeds.

Reply:
Updated! Thanks!